### PR TITLE
fix bug #1097

### LINKED ISSUE #1097: Edgemesh panic
**What happened**:
image

**What you expected to happen**:
Judge the slice.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeEdge version:
- Hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by arcanique**
**What type of PR is this?**
/kind bug
 
 **What this PR does / why we need it**:
fix bug #1097

**Which issue(s) this PR fixes**:
Fixes #1097 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note

```

### CODE REVIEW

File: edgemesh/pkg/proxy/proxy.go [hunk 1]
```diff
@@ -544,11 +545,14 @@ func (c *conntrack) processServerProxy() {
 
 //isL4Proxy Determine whether to use L4 proxy
 func IsL4Proxy(svc *v1.Service) bool {
+	if svc.Spec.Ports == nil || len(svc.Spec.Ports) == 0 {
```
Comment (fisherxu) [inline edgemesh/pkg/proxy/proxy.go]: Have `len(svc.Spec.Ports) == 0` this one is enough 
```suggestion
	if len(svc.Spec.Ports) == 0 {
```

Comment (fisherxu) [reply edgemesh/pkg/proxy/proxy.go]: And switch suggest to:
```suggestion
	switch port.Name {
	case "websocket", "grpc", "https", "tcp":
		return true
	case "http", "udp":
		return false
	default:
		return true
	}
```

**Comment by fisherxu** [comment]
Thanks for quick fix, @arcanique ptal :)

File: edgemesh/pkg/proxy/proxy.go (hunk 1 above)
Comment (arcanique) [reply edgemesh/pkg/proxy/proxy.go]: Thanks for suggestions. please review again

**Comment by sids-b** [comment]
Thanks @arcanique 

/approve

**Pull request merged**